src/containerapp/azext_containerapp/java_component_decorator.py

```diff
@@ -65,52 +65,29 @@ def get_argument_max_replicas(self):
-                "maxReplicas": self.get_argument_max_replicas()
-            }
+    def get_argument_set_configurations(self):
+        return self.get_param("set_configurations")
 
-        if self.get_argument_configuration() is not None:
-            configuration_list = []
-            for pair in self.get_argument_configuration():
-                key_val = pair.split('=', 1)
-                if len(key_val) != 2:
-                    raise ValidationError("Java configuration must be in format \"<propertyName>=<value> <propertyName>=<value> ...\".")
-                configuration_list.append({
-                    "propertyName": key_val[0],
-                    "value": key_val[1]
-                })
-            self.java_component_def["properties"]["configurations"] = configuration_list
+    def get_argument_replace_configurations(self):
+        return self.get_param("replace_configurations")
 
-    def create(self):
-        try:
-            return self.client.create(
-                cmd=self.cmd, resource_group_name=self.get_argument_resource_group_name(),
-                environment_name=self.get_argument_environment_name(), name=self.get_argument_java_component_name(),
-                java_component_envelope=self.java_component_def,
-                no_wait=self.get_argument_no_wait())
-        except Exception as e:
-            stringErr = str(e)
-            if "JavaComponentsNotAllowedForSubscription" in stringErr:
-                raise CLIInternalError("Java Components operations are not allowed for the subscription, please use 'az feature register --namespace  Microsoft.App --name JavaComponentsPreview' to register this feature.")
+    def get_argument_remove_configurations(self):
+        return self.get_param("remove_configurations")
 
-            handle_raw_exception(e)
+    def get_argument_remove_all_configurations(self):
+        return self.get_param("remove_all_configurations")
 
-    def update(self):
-        try:
-            return self.client.update(
-                cmd=self.cmd, resource_group_name=self.get_argument_resource_group_name(),
-                environment_name=self.get_argument_environment_name(), name=self.get_argument_java_component_name(),
-                java_component_envelope=self.java_component_def,
-                no_wait=self.get_argument_no_wait())
-        except Exception as e:
-            handle_raw_exception(e)
+    def set_configuration_with_legacy_way(self):
+        return self.get_argument_configuration() is not None
+
+    def set_configuration_with_new_way(self):
+        return self.get_argument_set_configurations() is not None or self.get_argument_replace_configurations() is not None or self.get_argument_remove_configurations() is not None or self.get_argument_remove_all_configurations() is not None
+
+    def validate_configurations(self):
+        if self.set_configuration_with_legacy_way() and self.set_configuration_with_new_way():
```

Comment:
Please add test for this validation:
For example:
```

        with self.assertRaisesRegex(CLIInternalError,
                                    "please unbind the Java Component first"):
            self.cmd('containerapp env java-component config-server-for-spring delete -g {} -n {} --environment {} --yes'.format(resource_group, config_name, env_name))
```

Reply:
This test is added